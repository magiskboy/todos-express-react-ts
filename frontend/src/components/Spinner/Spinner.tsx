import React from 'react';
import s from './Spinner.module.css';

export const Spinner: React.FC = () => {
  return <div className={s.spinner}></div>;
};