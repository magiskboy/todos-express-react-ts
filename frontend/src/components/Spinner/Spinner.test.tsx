import { render } from "@testing-library/react";
import { Spinner } from "./Spinner";

describe("Spinner component", () => {
  it("renders without crashing", () => {
    const { container } = render(<Spinner />);
    expect(container).toMatchSnapshot();
  });

  it("renders the spinner element", () => {
    const { getByTestId } = render(<Spinner />);
    const spinnerElement = getByTestId("spinner");
    expect(spinnerElement).toBeDefined();
  });
});
