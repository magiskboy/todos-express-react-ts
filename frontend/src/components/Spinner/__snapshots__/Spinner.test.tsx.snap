// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component renders without crashing 1`] = `
<div>
  <div
    class="spinner"
    data-testid="spinner"
  />
</div>
`;
